### PR TITLE
Adds VISION.md document

### LINKED ISSUE #5: Vision Document
This doesn't have to be long or complex, but a few sentences in `VISION.md` describing what the overall goal of Harvey is will help a lot once folks start contributing back.

### PR DESCRIPTION
**Pull request by sunshinejr**
First draft of the VISION document. Fixes #5. Let me know how it looks!

### CODE REVIEW

File: VISION.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Harvey Vision
+
+For a long time at Moya we were struggling with migrating the codebase to be `SPM-test` compatible. We acknowledge that we need to replace our current network stubbing library,
+[OHHTTPStubs](https://github.com/AliSoftware/OHHTTPStubs), with something that is SPM
+compatible, written in Swift and actively maintained. We searched for replacements, but
+nothing met our criteria, so we decided to create our own library under Moya organization.
```
Comment (SD10) [inline VISION.md]: under Moya organization -> under the Moya organization

File: VISION.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Harvey Vision
+
+For a long time at Moya we were struggling with migrating the codebase to be `SPM-test` compatible. We acknowledge that we need to replace our current network stubbing library,
```
Comment (sunshinejr) [inline VISION.md]: acknowledge -> acknowledged

File: VISION.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+[OHHTTPStubs](https://github.com/AliSoftware/OHHTTPStubs), with something that is SPM
+compatible, written in Swift and actively maintained. We searched for replacements, but
+nothing met our criteria, so we decided to create our own library under Moya organization.
+
+# Project goals
+- provide network response stubbing based on request properties (url/method etc.)
+- provide simple, yet extensible, API
+- satisfy whatever needs we have for network stubbing over at Moya
```
Comment (SD10) [inline VISION.md]: "satisfy the network stubbing needs expressed by Moya users" ?

File: VISION.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+- find out if we can leverage some stubbing work off of Moya
```
Comment (SD10) [inline VISION.md]: this one confuses me because I feel its vague. are you talking about decoupling stubbing from Moya?

Comment (sunshinejr) [reply VISION.md]: Hmm, yeah maybe wrong wording. This one is actually a R&D type of thing, I wanted to do a research and find out what we can do and what we can't by connecting Harvey and Moya using plugins. I didn't really think of decoupling stubbing from Moya, as Harvey would use swizzling and Moya has a first class stubbing mechanism that does not need swizzling right now (other than for testing purposes).

Comment (SD10) [reply VISION.md]: Great, thanks for clarifying!

"determine the extent to which Harvey can be bootstrapped using Moya's first class stubbing API" -- kind of wordy but 😕

File: VISION.md (hunk 3 above)
Comment (ashfurrow) [reply VISION.md]: @SD10 I'm not sure I agree – Moya already supports network response stubbing, so users will likely be using either Moya or Harvey. To be honest, I kind of like the straightforwardness of something like "satisfy the testing needs of Moya" – what do you think?

File: VISION.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+- fully SPM-compatible (`build`/`test`)
+- actively maintained/updated to newest Moya/Swift versions
```
Comment (ashfurrow) [inline VISION.md]: Maybe something like "leverage the [Moya community](https://github.com/orgs/Moya/people) to keep Harvey updated to the latest Moya/Swift versions" ?

Comment (SD10) [reply VISION.md]: Makes sense @ashfurrow. I tend to be prolix. I just didn't like using the words "whatever" and "over at Moya"

File: VISION.md (hunk 3 above)
Comment (SD10) [reply VISION.md]: Makes sense @ashfurrow. I tend to be prolix. I just didn't like using the words "whatever" and "over at Moya"

**Comment by sunshinejr** [comment]
Thanks you very much @ashfurrow @SD10! I've adjusted the document, please let me know how it looks now 👍

**Pull request merged**